search_service/proxy/atlas.py

```diff
@@ -0,0 +1,230 @@
+
+from search_service.models.search_result import SearchResult
+from search_service.models.table import Table
+from search_service.proxy import BaseProxy
+from search_service.proxy.statsd_utilities import timer_with_counter
+
+DEFAULT_PAGE_SIZE = 10
+LOGGING = logging.getLogger(__name__)
+
+
+class AtlasProxy(BaseProxy):
+    TABLE_ENTITY = app.config['ATLAS_TABLE_ENTITY']
+    DB_ATTRIBUTE = app.config['ATLAS_DB_ATTRIBUTE']
+    NAME_ATTRIBUTE = app.config['ATLAS_NAME_ATTRIBUTE']
+    ATTRS_KEY = 'attributes'
+    REL_ATTRS_KEY = 'relationshipAttributes'
+
+    """
+    AtlasSearch connection handler
+    """
+    atlas: Atlas
+
+    def __init__(self, *,
+                 host: str = None,
+                 index: str = None,
+                 user: str = '',
+                 password: str = '',
+                 atlas_client: Atlas = None,
+                 page_size: int = DEFAULT_PAGE_SIZE) -> None:
+        if atlas_client:
+            self.atlas = atlas_client
+        else:
+            self.atlas = self._create_client_from_credentials(host=host,
+                                                              user=user,
+                                                              password=password)
+
+        self.index = index
+        self.page_size = page_size
+
+    @staticmethod
+    def _create_client_from_credentials(*,
+                                        host: str = None,
+                                        user: str = '',
+                                        password: str = '') -> Atlas:
+        """
+        Construct Atlas client that connects to cluster at {host} and {port}
+        and authenticates using {user} and {password}
+        :return: Atlas client object
+        """
+        return Atlas(host, username=user, password=password)
+
+    @staticmethod
+    def _entities(collections: EntityCollection) -> List[Entity]:
+        """
+        Helper method for flattening all collections from {collections}
+        :return: list of all entities
+        """
+        entities: List[Entity] = []
+        for collection in collections:
+            entities.extend(collection.entities)
+        return entities
+
+    def _parse_results(self, response: EntityCollection) -> List[Table]:
+        """
+        based on an atlas {response} with table entities, we map the required information
+        :return: list of tables
+        """
+        table_results = []
+        ids = list()
+        for hit in response:
+            ids.append(hit.guid)
+        # receive all entities
+        entities = self._entities(self.atlas.entity_bulk(guid=ids))
+        db_ids = []
+        for entity in entities:
+            relations = entity.relationshipAttributes
+            database = relations.get(self.DB_ATTRIBUTE)
+            if database:
+                db_ids.append(database['guid'])
+
+        # request databases
+        dbs_list = self._entities(self.atlas.entity_bulk(guid=db_ids)) if len(db_ids) > 0 else []
```

Comment:
Nanne I think we don't need to do this here. as per [documentation of api](https://atlas.apache.org/api/v2/resource_EntityREST.html#resource_EntityREST_getByGuids_GET) `entity_bulk` should return `referredEntities` information which can be used for the database here. Did you try this, could save loops and extra query ?

Reply:
unfortunately it does not provide all attributes from the database.

Reply:
I made sure all the requests are done via `entity_bulk` requests